From ca5630eddd4cc8846b3ffc35d44acc77f6c9aebd Mon Sep 17 00:00:00 2001
From: Andriy Grytsenko <andrej@rep.kiev.ua>
Date: Sat, 24 Nov 2012 21:23:50 +0200
Subject: [PATCH 11/11] Desktop items layout haven't respected reserved space
 on monitor.

If panels reserve space on monitor then items shouldn't be placed
on that space. That was broken, the commit fixes that.
---
 src/desktop.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/desktop.c b/src/desktop.c
index 66201f7..6725313 100644
--- a/src/desktop.c
+++ b/src/desktop.c
@@ -417,8 +417,8 @@ static void layout_items(FmDesktop* self)
             else
             {
 _next_position:
-                item->x = x;
-                item->y = y;
+                item->x = self->working_area.x + x;
+                item->y = self->working_area.y + y;
                 calc_item_size(self, item, icon);
                 y += self->cell_h;
                 if(y > bottom)
@@ -448,8 +448,8 @@ _next_position:
             else
             {
 _next_position_rtl:
-                item->x = x;
-                item->y = y;
+                item->x = self->working_area.x + x;
+                item->y = self->working_area.y + y;
                 calc_item_size(self, item, icon);
                 y += self->cell_h;
                 if(y > bottom)
-- 
1.8.0.1

